src/bloggy_frontend/src/pages/Posts.jsx

```diff
@@ -1,41 +1,49 @@
-import React, { useEffect, useState } from 'react'
-import { Container, Row, Col, Card, Button } from 'react-bootstrap'
+import React from 'react'
+import { Container, Row, Col, Card, Button, Spinner } from 'react-bootstrap'
+import { useQuery } from '@tanstack/react-query'
 import { bloggy_backend } from '../../../declarations/bloggy_backend'
 import { Link } from 'react-router-dom'
+import { useAuth } from '../context/AuthContext'
+import { QUERY_KEYS } from '../constants/queryKeys'
 
-const Posts = () => {
-  const [posts, setPosts] = useState([])
-  const [loading, setLoading] = useState(true)
-  const [error, setError] = useState('')
-
-  useEffect(() => {
-    const fetchPosts = async () => {
-      try {
-        const fetchedPosts = await bloggy_backend.viewPosts()
-        setPosts(fetchedPosts)
-      } catch (err) {
-        console.error('Error fetching posts:', err)
-        setError('Failed to fetch posts.')
-      } finally {
-        setLoading(false)
-      }
-    }
+const fetchPosts = async () => {
+  const posts = bloggy_backend.viewPosts()
+  return posts
```

Comment:
we can just `return bloggy_backend.viewPost()` no need to store it to a temp variable
```suggestion
return loggy_backend.viewPosts()
```